### DISCUSSION TITLE
Using automatic authentication refresh in non EKS clusters

### DISCUSSION THREAD

**Discussion by tomikonio** [upbound/provider-aws#1126] [Q&A]
Hi,
I am running Crossplane on a non AWS cluster (e.g. local).
I would like to use the [Token provider configuration with automatic authentication refresh for IAM Identity Center](https://docs.aws.amazon.com/cli/latest/userguide/sso-configure-profile-token.html) method for authentication with AWS.
According to the [docs](https://docs.upbound.io/providers/provider-aws/authentication/), it seems the only way to connect to AWS from a non EKS cluster is to use AWS access keys (secret) method.
Is this really the case? Is there another way to do this?

**Comment by laseanb** [comment]
I'm also curious as to whether this is the case. Are access keys the only supported way to authenticate outside of a non-eks cluster??

**Comment by laseanb** [comment]
This doesn't seem like your specific use case @tomikonio , but just wanted to share that I just recently found that I could get the provider to assume the role given to the node by creating a `ProviderConfig` with `spec.credentials.source: IRSA` even though I DON'T have IRSA setup. (I'm guessing that this allows the AWS CLI/SDK to attempt authentication > fail > fallback to the node role)
